src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -255,9 +244,10 @@ object TaskReplaceActor extends StrictLogging {
     def canStartNewInstances: Boolean = minHealthy < maxCapacity || consideredHealthyInstancesCount - nrToKillImmediately < maxCapacity
     assume(canStartNewInstances, "must be able to start new instances")
 
-    val leftCapacity = math.max(0, maxCapacity - totalInstancesRunning)
+    val leftCapacity = math.max(0, maxCapacity - totalInstancesRunning + nrToKillImmediately)
```

Comment:
Would it be an issue to remove this? Will the deployment take longer to complete? Will some tests fail?

Reply:
Now that CI is fully working I'll have a look if I can remove this